### PR TITLE
add package tor

### PR DESCRIPTION
**Pull request by starius**
Tor is free software and an open network that helps you defend
against traffic analysis, a form of network surveillance that
threatens personal freedom and privacy, confidential business
activities and relationships, and state security.

https://www.torproject.org/

### CODE REVIEW

File: src/tor.mk [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# This file is part of MXE.
+# See index.html for further information.
+
+https://www.torproject.org/dist/tor-0.2.6.10.tar.gz
```
Comment (TimothyGu) [inline src/tor.mk]: huh?

**Comment by TimothyGu** [comment]
Does Tor have a library?

File: src/tor.mk (hunk 1 above)
Comment (starius) [inline src/tor.mk]: fixed

**Comment by starius** [comment]
Unfortunately there is no Tor library (at least official). That is why all applications using Tor (e.g., Tor Browser) are bundeled with Tor executable.

**Comment by TimothyGu** [comment]
So… this package isn't a library? I'm afraid that we might not accept that: http://mxe.cc/#creating-packages

> The package should be a free software library that is really used by one of your applications.

**Comment by starius** [comment]
Cross-compiling tools using MXE is also valueble. Can we create a separate place for packages that are not libraries?

**Comment by TimothyGu** [comment]
tor isn't strictly a tool don't you think?

**Comment by starius** [comment]
Tor is a tool working in background :-)
Even if it is not a library literally, it is used as if it is a library by many programs and libraries. For example, [Stem](https://stem.torproject.org/) is Python library using Tor.

Personally I would preffer to use Tor built in Linux (with MXE) instead of built in Windows, because I don't trust build toolchains operating in Windows. That is why I added it to MXE: to have a reliable place for getting Tor for Windows.

**Comment by starius** [comment]
By the way, there are packages in MXE which are not libraries but programs:
 * wget
 * openmp-validation
 * nsis

**Comment by tonytheodore** [comment]
> there are packages in MXE which are not libraries but programs:

MXE is targeted at developers and there's a class of handy tools (gdb and sqlite in addition to the ones you mention) that make it easier to test and debug your newly built app. I can see the argument that you'd want `tor` on hand to make sure your app can be easily configured to use it reliably.

> because I don't trust build toolchains operating in Windows

Not even the official ones from a project with a mindset like Tor? The logical conclusion of that line of reason is to abandon windows altogether, including developing for it. Too philosophical and not really relevant to it's inclusion.

> That is why I added it to MXE: to have a reliable place for getting Tor for Windows.

That is probably the exact reason we don't build most binaries - we don't want to become an unofficial source.

**Comment by starius** [comment]
> Not even the official ones from a project with a mindset like Tor? The logical conclusion of that line of reason is to abandon windows altogether, including developing for it. Too philosophical and not really relevant to it's inclusion.

I want to built Tor manually instead of downloading a pre-built binary even if it was built by official maintainer. Building it on Windows is not reliable, as I said. And one more reason: builing on Windows is less convenient (at least for me) than just typing `make tor`. Using Tor on Windows is still needed and it is secure if Windows itself is constrained (for example it lives in VBox).

> That is probably the exact reason we don't build most binaries - we don't want to become an unofficial source.

I wish there is a source like MXE for tools I have to use on Windows. Tor is one of them. Building them with MXE seems more secure (and convenient!) than building them on Windows or downloading binaries (often bundeled with viruses).

**Comment by tonytheodore** [comment]
> I wish there is a source like MXE for tools I have to use on Windows.

Indeed! Unfortunately "user" tools are (currently) outside the scope of MXE, but something like [MSYS2](https://msys2.github.io/) may be along the lines of what you're looking for.

I can see great value in a downstream MXE project that builds tools, demonstrates use-cases, maybe builds some complex apps (say VLC)...

**Pull request closed without merging**